src/Rings/mpoly-ideals.jl

```diff
@@ -1250,6 +1250,36 @@ julia> codim(I)
 """
 codim(I::MPolyIdeal) = nvars(base_ring(I)) - dim(I)
 
+#######################################################
+#######################################################
+@doc raw"""
+    degree(I::MPolyIdeal)
+
+Return the degree of `I` if it is homogeneous. Otherwise return the degree
+of its `homogenization`.
+
+The degree of a homogeneous ideal `I` is the number of intersection points
+of its projective variety with a generic linear subspace of complementary
+dimension (counted with multiplicities). See also [MS21](@cite).
+
+# Examples
+```jldoctest
+julia> R, (x, y, z) = polynomial_ring(QQ, ["x", "y", "z"])
+(Multivariate polynomial ring in 3 variables over QQ, QQMPolyRingElem[x, y, z])
+
+julia> I = ideal(R, [y-x^2, x-z^3])
+ideal(-x^2 + y, x - z^3)
+
+julia> degree(I)
+6
+```
+"""
+@attr Int function degree(I::MPolyIdeal)
+  J = homogenization(I, "_h")
+  A, _ = quo(base_ring(J), J)
+  return Int(degree(A))
```

Comment:
Suggested change:  

  is_standard_graded(base_ring(I)) || (I = homogenization(I, "_h")) 
  return Int(degree(HilbertData(I)))

In this way, in the homogeneous case, only one GB calculation is needed.
I